test/parallel/test-tls-connect-simple.js

```diff
@@ -1,58 +1,40 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
+const common = require('../common');
+const assert = require('assert');
 
 if (!common.hasCrypto) {
   common.skip('missing crypto');
   return;
 }
-var tls = require('tls');
+const tls = require('tls');
 
-var fs = require('fs');
+const fs = require('fs');
 
-var clientConnected = 0;
-var serverConnected = 0;
-var serverCloseCallbacks = 0;
-var serverCloseEvents = 0;
-
-var options = {
+const options = {
   key: fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
   cert: fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem')
 };
 
-var server = tls.Server(options, function(socket) {
-  if (++serverConnected === 2) {
```

Comment:
I don't think you can drop this check. It changes the behavior. The server will be closed the first time instead of the second.